app/src/main/java/com/sdpteam/connectout/MapActivity.java

```diff
@@ -0,0 +1,94 @@
+package com.sdpteam.connectout;
+
+import android.os.Bundle;
+import android.view.View;
+import android.widget.Button;
+import android.widget.Toast;
+
+import androidx.annotation.Nullable;
+import androidx.fragment.app.FragmentActivity;
+import androidx.lifecycle.Observer;
+import androidx.lifecycle.ViewModelProvider;
+
+import com.google.android.gms.maps.CameraUpdateFactory;
+import com.google.android.gms.maps.GoogleMap;
+import com.google.android.gms.maps.OnMapReadyCallback;
+import com.google.android.gms.maps.SupportMapFragment;
+import com.google.android.gms.maps.model.LatLng;
+import com.google.android.gms.maps.model.Marker;
+import com.google.android.gms.maps.model.MarkerOptions;
+
+import java.util.List;
+
+public class MapActivity extends FragmentActivity implements OnMapReadyCallback {
+
+    private GoogleMap mMap;
+    private Button refreshButton;
+
+    //container object for the map used for the map lifecycle management and UI capabilities
+    private SupportMapFragment mapFragment;
+    private MapViewModel mapViewModel;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+
+        setContentView(R.layout.activity_map);
+
+        // Obtain the SupportMapFragment and get notified when the map is ready to be used.
+        mapFragment = (SupportMapFragment) getSupportFragmentManager()
+                .findFragmentById(R.id.map);
+        mapFragment.getMapAsync(this);
+
+        mapViewModel = new ViewModelProvider(this).get(MapViewModel.class);
+        mapViewModel.init(new MapModel());
+
+        mapViewModel.getEventList().observe(this, eventList -> showNewMarkerList(eventList));
+
+        refreshButton = (Button) findViewById(R.id.refresh_button);
+        refreshButton.setOnClickListener(view -> showNewMarkerList(mapViewModel.refreshEventList().getValue()));
+    }
+
+    private void showNewMarkerList(List<Event> eventList) {
+        if (mMap == null) {
+            return;
```

Comment:
I think you should throw an `IllegalStateException` instead of not saying anything

Reply:
That's a very good point